### PR TITLE
ModificationDateTime column + number of queries limit

### PR DESCRIPTION
**Pull request by leonPep42**

### CODE REVIEW

File: server-side/installation.ts [hunk 1]
```diff
@@ -70,7 +70,7 @@ export async function upgrade(client: Client, request: Request): Promise<any> {
 			await pnsService.subscribeToRelationsUpdate();
         }
 
-		if (request.body.FromVersion && semver.compare(request.body.FromVersion, '1.4.9') < 0)
+		if (request.body.FromVersion && semver.compare(request.body.FromVersion, '1.4.20') < 0)
```
Comment (saargt) [inline server-side/installation.ts:73]: After setting VarSettings you must also apply these VarSettings to any existing queries.

File: server-side/services/varSettings.service.ts [hunk 2]
```diff
@@ -26,9 +26,17 @@ export class VarSettingsService {
     }
 
 	async upsertVarSettings(varSettings: VarSettingsObject): Promise<DIMXObject[]> {
+		if (parseInt(varSettings.DaysLimit) > 180) {
+			throw new Error('Days limit value cannot exceed 180');
```
Comment (saargt) [inline server-side/services/varSettings.service.ts:30]: I think you really have two functions here - 1) Update KMS parameters 2) Set VarSettings on queries.
upsertVarSettings should call these two functions.

File: server-side/services/varSettings.service.ts [hunk 3]
```diff
@@ -26,9 +26,17 @@ export class VarSettingsService {
     }
 
 	async upsertVarSettings(varSettings: VarSettingsObject): Promise<DIMXObject[]> {
+		if (parseInt(varSettings.DaysLimit) > 180) {
```
Comment (saargt) [inline server-side/services/varSettings.service.ts:29]: Please set modifiers on the methods.

File: server-side/installation.ts (hunk 1 above)
Comment (saargt) [reply server-side/installation.ts:73]: I thing this entire logic should be placed inside the VarSettingsService

File: server-side/installation.ts [hunk 4]
```diff
@@ -44,7 +44,9 @@ export async function install(client: Client, request: Request): Promise<any> {
 export async function uninstall(client: Client, request: Request): Promise<any> {
     try {
         const service = new UtilitiesService(client);
+		const varSettingsService = new VarSettingsService(client);
         await service.papiClient.post(`/addons/data/schemes/${DATA_QUREIES_TABLE_NAME}/purge`);
```
Comment (saargt) [inline server-side/installation.ts:48]: Any reason you handle the purge of the data queries schema by yourself here? Unless you rely on some PNSs for deleting items or the schema itself, I think you can just rely on ADAL to do it for you automatically on uninstall.

File: server-side/services/varSettings.service.ts [hunk 5]
```diff
@@ -25,54 +25,82 @@ export class VarSettingsService {
 		this.queryService = new QueryService(client);
     }
 
-	async upsertVarSettings(varSettings: VarSettingsObject): Promise<DIMXObject[]> {
+	public async initializeVarSettings(): Promise<void> {
+		await this.upsertVarSettingsRelation();
+		await this.setDefaultVarSettings();
+	}
```
Comment (saargt) [inline server-side/services/varSettings.service.ts:31]: I think I'm missing where you actually set the new var settings on existing queries. Where's that done? (On upgrade you set these defualts, but they also have to be applied to any existing queries, right?)

File: server-side/services/varSettings.service.ts (hunk 3 above)
Comment (saargt) [reply server-side/services/varSettings.service.ts:43]: @leonPep42 ⬆️

File: server-side/services/varSettings.service.ts (hunk 5 above)
Comment (leonPep42) [reply server-side/services/varSettings.service.ts:31]: setDefaultVarSettings calls upsertVarSettings which does it

**Pull request merged**